src/super_context.ml

```diff
@@ -33,6 +33,245 @@ module Env_node = struct
     }
 end
 
+module Var : sig
+  module Kind : sig
+    type t =
+      | Values of Value.t list
+      | Project_root
+      | First_dep
+      | Deps
+      | Targets
+
+    val to_value_no_deps_or_targets : t -> scope:Scope.t -> Value.t list option
+  end
+
+  module Form : sig
+    type t =
+      | Exe
+      | Dep
+      | Bin
+      | Lib
+      | Libexec
+      | Lib_available
+      | Version
+      | Read
+      | Read_strings
+      | Read_lines
+      | Path_no_dep
+  end
+
+  type 'a t =
+    | Nothing    of 'a
```

Comment:
the name `Nothing` was meant to mean `no information`. `Nothing of 'a` looks a bit odd, maybe just `T of 'a`?

Reply:
I changed it to `No_info`. `T` is already our convention for open types.